vllm/entrypoints/openai/protocol.py

```diff
@@ -476,6 +500,12 @@ def to_sampling_params(
                 json_schema = self.response_format.json_schema
                 assert json_schema is not None
                 self.guided_json = json_schema.json_schema
+            elif self.response_format.type == "structural_tag":
+                structural_tag = self.response_format
+                assert structural_tag is not None and isinstance(
+                    structural_tag, StructuralTagResponseFormat)
+                s_tag_obj = structural_tag.model_dump(by_alias=True)
+                self.structural_tag = json.dumps(s_tag_obj)
```

Comment:
```suggestion
                self.structural_tag = structural_tag.model_dump_json(by_alias=True)
```

Then you don't have to use json here :)